src/partition.ml

```diff
@@ -0,0 +1,94 @@
+open Encoding
+module H_contents = Irmin.Hash.Typed (Hash) (Contents)
+
+type entry = { name : string; kind : [ `Node | `Contents ]; hash : Hash.t }
+type enc_entry = { eencoding : string; entry : entry }
+type pointer = { index : int; hash : Hash.t }
+type enc_pointer = { pencoding : string; pointer : pointer }
+
+type vs =
+  | Empty
+  | Values of enc_entry list
+  | Tree of {
+      depth : int;
+      length : int;
+      pointers : (enc_pointer * enc_vs) list;
+    }
+
+and enc_vs = { vsencoding : string; vs : vs }
+
+let pre_hash t v =
+  let buf = Buffer.create 13 in
+  let pre_hash = Irmin.Type.(unstage (pre_hash t)) in
+  pre_hash v (Buffer.add_string buf);
+  Buffer.contents buf
+
+let compute_hash s = H_contents.hash (Bytes.of_string s)
+let pp_hash fmt h = Irmin.Type.(pp Encoding.Hash.t) fmt h
+let char_of_kind = function `Contents -> '\000' | `Node -> '\001'
```

Comment:
it seems to be the other way around here (https://hackmd.io/@samoht/tezos-context-hash#Inode-values)